### ISSUE TITLE
Failure with kubernetes/pause image on fresh minions

### ISSUE THREAD

**Issue by jdojff** [pires/kubernetes-vagrant-coreos-cluster#46]
Hi, first time I try to deploy some pods on to the cluster I receive failures in endless loop. This is until I ssh into every minion and run ```docker pull kubernetes/pause``` manually.
```
Wed, 18 Mar 2015 14:13:55 +0000   Wed, 18 Mar 2015 14:13:55 +0000   1                   myservice      BoundPod            implicitly required container POD   pulled              {kubelet 172.17.8.103}   Successfully pulled image "kubernetes/pause:latest"
Wed, 18 Mar 2015 14:13:55 +0000   Wed, 18 Mar 2015 14:13:55 +0000   1                   myservice      BoundPod            implicitly required container POD   pulled              {kubelet 172.17.8.103}   Successfully pulled image "kubernetes/pause:latest"
Wed, 18 Mar 2015 14:13:55 +0000   Wed, 18 Mar 2015 14:13:55 +0000   1                   myservice      BoundPod                                                failedSync          {kubelet 172.17.8.103}   Error syncing pod, skipping: no such image
Wed, 18 Mar 2015 14:13:55 +0000   Wed, 18 Mar 2015 14:13:55 +0000   1                   myservice      BoundPod            implicitly required container POD   failed              {kubelet 172.17.8.103}   Failed to create docker container with error: no such image
Wed, 18 Mar 2015 14:14:05 +0000   Wed, 18 Mar 2015 14:14:05 +0000   1                   myservice      BoundPod                                                failedSync          {kubelet 172.17.8.103}   Error syncing pod, skipping: Authentication is required.
Wed, 18 Mar 2015 14:14:05 +0000   Wed, 18 Mar 2015 14:14:05 +0000   1                   myservice      BoundPod            implicitly required container POD   failed              {kubelet 172.17.8.103}   Failed to pull image "kubernetes/pause:latest"
```

**Comment by AntonioMeireles** [comment]
@jdojff could you plz on a cluster being hit by this past here the output in the master of `journalctl -u docker-cache --no-pager --lines=1000` and on any of the minions of `journalctl -u docker --no-pager --lines=1000` ?

**Comment by jdojff** [comment]
Sure, master: http://pastebin.com/DwK3DqKG
node-01: http://pastebin.com/cYLDVpbH
node-02: http://pastebin.com/D25qEWKy

**Comment by pires** [comment]
@jdojff working on a hotfix. Please bear with me.

**Comment by pires** [comment]
@jdojff care to try it out now and give feedback, please?

**Comment by jdojff** [comment]
No, the issue is still there. I think these recent big changes hadn't do anything with it. I already had this issue one-two weeks ago when trying this cluster configuration for the first time.

**Comment by pires** [comment]
@jdojff when are you available for a debugging session?

**Comment by AntonioMeireles** [comment]
interested in the *digging* expedition too... :angel:

**Comment by pires** [comment]
Just booted a new Kubernetes cluster, deployed Elasticsearch cluster and..
```
core@node-01 ~ $ docker ps
CONTAINER ID        IMAGE                        COMMAND                CREATED             STATUS              PORTS                              NAMES
be3a8f2ff813        pires/elasticsearch:master   "/usr/bin/runsvdir -   44 minutes ago      Up 44 minutes                                          k8s_elasticsearch-master.7f022321_elasticsearch-master-fplln_default_bb2d229b-cd92-11e4-9294-080027ddceed_66cbd6ac
7fdfa48c7c77        kubernetes/pause:go          "/pause"               About an hour ago   Up About an hour                                       k8s_POD.c57df10a_elasticsearch-master-fplln_default_bb2d229b-cd92-11e4-9294-080027ddceed_c21f4fa0
161d2bb56b7e        google/cadvisor:0.10.1       "/usr/bin/cadvisor -   About an hour ago   Up About an hour    8080/tcp, 0.0.0.0:4194->4194/tcp   cadvisor
```

The only difference I spot is the `kubernetes/pause` tag, which in my case is `go` while in your case is `latest`.

**Comment by jdojff** [comment]
I'm available now. Ok, so I have checked out fresh ```pires/kubernetes-vagrant-coreos-cluster``` and ```pires/kubernetes-elasticsearch-cluster``` repos and tried with your example.

First thing to get my attention is, when starting ```vagrant up master``` and ```vagrant up``` I get ```Could not open a connection to your authentication agent.``` in each node: http://pastebin.com/Sz8uHt98 Not sure if this has something to do with it or not.

Moving on.. this time I did not received any failures:
```
[root@localhost kubernetes-elasticsearch-cluster]# kubectl get pods
POD                          IP                  CONTAINER(S)           IMAGE(S)                     HOST                        LABELS                                       STATUS
elasticsearch-data-oh43e     10.244.16.4         elasticsearch-data     pires/elasticsearch:data     172.17.8.102/172.17.8.102   component=elasticsearch,role=data            Pending
elasticsearch-lb-gziey       10.244.16.3         elasticsearch-lb       pires/elasticsearch:lb       172.17.8.102/172.17.8.102   component=elasticsearch,role=load-balancer   Pending
elasticsearch-master-fplln   10.244.21.3         elasticsearch-master   pires/elasticsearch:master   172.17.8.103/172.17.8.103   component=elasticsearch,role=master          Pending
[root@localhost kubernetes-elasticsearch-cluster]# kubectl get events
FIRSTSEEN                         LASTSEEN                          COUNT               NAME                         KIND                SUBOBJECT                           REASON              SOURCE                   MESSAGE
Thu, 19 Mar 2015 07:30:43 +0000   Thu, 19 Mar 2015 07:30:43 +0000   1                   172.17.8.102                 Minion                                                  starting            {kubelet 172.17.8.102}   Starting kubelet.
Thu, 19 Mar 2015 07:31:09 +0000   Thu, 19 Mar 2015 07:31:09 +0000   1                   172.17.8.103                 Minion                                                  starting            {kubelet 172.17.8.103}   Starting kubelet.
Thu, 19 Mar 2015 07:35:20 +0000   Thu, 19 Mar 2015 07:35:20 +0000   1                   elasticsearch-master-fplln   Pod                                                     scheduled           {scheduler }             Successfully assigned elasticsearch-master-fplln to 172.17.8.103
Thu, 19 Mar 2015 07:35:20 +0000   Thu, 19 Mar 2015 07:35:20 +0000   1                   elasticsearch-lb-gziey       Pod                                                     scheduled           {scheduler }             Successfully assigned elasticsearch-lb-gziey to 172.17.8.102
Thu, 19 Mar 2015 07:35:22 +0000   Thu, 19 Mar 2015 07:35:22 +0000   1                   elasticsearch-data-oh43e     Pod                                                     scheduled           {scheduler }             Successfully assigned elasticsearch-data-oh43e to 172.17.8.102
Thu, 19 Mar 2015 07:35:30 +0000   Thu, 19 Mar 2015 07:35:30 +0000   1                   elasticsearch-lb-gziey       Pod                 implicitly required container POD   started             {kubelet 172.17.8.102}   Started with docker id d705a4ffe6d6840b8f111eb463deaefc2c03a19960e352e820e3de8cf05b2a0d
Thu, 19 Mar 2015 07:35:30 +0000   Thu, 19 Mar 2015 07:35:30 +0000   2                   elasticsearch-master-fplln   Pod                 implicitly required container POD   pulled              {kubelet 172.17.8.103}   Successfully pulled image "kubernetes/pause:latest"
Thu, 19 Mar 2015 07:35:30 +0000   Thu, 19 Mar 2015 07:35:30 +0000   1                   elasticsearch-master-fplln   Pod                 implicitly required container POD   created             {kubelet 172.17.8.103}   Created with docker id 89b5e595eddb8dae0b7443cad3b6c4402a25aaf2479fa00433766dd76d2679de
Thu, 19 Mar 2015 07:35:30 +0000   Thu, 19 Mar 2015 07:35:30 +0000   1                   elasticsearch-master-fplln   Pod                 implicitly required container POD   started             {kubelet 172.17.8.103}   Started with docker id 89b5e595eddb8dae0b7443cad3b6c4402a25aaf2479fa00433766dd76d2679de
Thu, 19 Mar 2015 07:35:30 +0000   Thu, 19 Mar 2015 07:35:30 +0000   2                   elasticsearch-lb-gziey       Pod                 implicitly required container POD   pulled              {kubelet 172.17.8.102}   Successfully pulled image "kubernetes/pause:latest"
Thu, 19 Mar 2015 07:35:30 +0000   Thu, 19 Mar 2015 07:35:30 +0000   2                   elasticsearch-data-oh43e     Pod                 implicitly required container POD   pulled              {kubelet 172.17.8.102}   Successfully pulled image "kubernetes/pause:latest"
Thu, 19 Mar 2015 07:35:30 +0000   Thu, 19 Mar 2015 07:35:30 +0000   1                   elasticsearch-data-oh43e     Pod                 implicitly required container POD   created             {kubelet 172.17.8.102}   Created with docker id e645cc14dad2908104eaea850fbdbba26fafafea714865420c0827933f791d8f
Thu, 19 Mar 2015 07:35:30 +0000   Thu, 19 Mar 2015 07:35:30 +0000   1                   elasticsearch-data-oh43e     Pod                 implicitly required container POD   started             {kubelet 172.17.8.102}   Started with docker id e645cc14dad2908104eaea850fbdbba26fafafea714865420c0827933f791d8f
Thu, 19 Mar 2015 07:35:30 +0000   Thu, 19 Mar 2015 07:35:30 +0000   1                   elasticsearch-lb-gziey       Pod                 implicitly required container POD   created             {kubelet 172.17.8.102}   Created with docker id d705a4ffe6d6840b8f111eb463deaefc2c03a19960e352e820e3de8cf05b2a0d
```
But still none of the pods were deployed. After waiting for 10 minutes, I have ran ```docker pull kubernetes/pause``` on ```node-01``` and it immediately pulled and started elastic search containers. 
```
core@node-01 ~ $ docker ps
CONTAINER ID        IMAGE                    COMMAND                CREATED             STATUS              PORTS                              NAMES
e645cc14dad2        kubernetes/pause:go      "/pause"               10 minutes ago      Up 10 minutes                                          k8s_POD.c57df10a_elasticsearch-data-oh43e_default_8062dbe1-ce0a-11e4-b598-080027ddceed_0e1ca766
d705a4ffe6d6        kubernetes/pause:go      "/pause"               10 minutes ago      Up 10 minutes                                          k8s_POD.bfd3151d_elasticsearch-lb-gziey_default_7f272ad1-ce0a-11e4-b598-080027ddceed_567158c8 
1045e15e6bae        google/cadvisor:0.10.1   "/usr/bin/cadvisor -   15 minutes ago      Up 15 minutes       8080/tcp, 0.0.0.0:4194->4194/tcp   cadvisor                                                                                      
core@node-01 ~ $ docker pull kubernetes/pause
Pulling repository kubernetes/pause
6c4579af347b: Download complete
511136ea3c5a: Download complete
e244e638e26e: Download complete
Status: Image is up to date for kubernetes/pause:latest
core@node-01 ~ $ docker ps
CONTAINER ID        IMAGE                    COMMAND                CREATED             STATUS              PORTS                              NAMES
33c3b211df41        pires/elasticsearch:lb   "/usr/bin/runsvdir -   12 seconds ago      Up 11 seconds                                          k8s_elasticsearch-lb.3ddd43a9_elasticsearch-lb-gziey_default_7f272ad1-ce0a-11e4-b598-080027ddceed_c9ac2df8
e645cc14dad2        kubernetes/pause:go      "/pause"               11 minutes ago      Up 11 minutes                                          k8s_POD.c57df10a_elasticsearch-data-oh43e_default_8062dbe1-ce0a-11e4-b598-080027ddceed_0e1ca766
d705a4ffe6d6        kubernetes/pause:go      "/pause"               11 minutes ago      Up 11 minutes                                          k8s_POD.bfd3151d_elasticsearch-lb-gziey_default_7f272ad1-ce0a-11e4-b598-080027ddceed_567158c8 
1045e15e6bae        google/cadvisor:0.10.1   "/usr/bin/cadvisor -   16 minutes ago      Up 16 minutes       8080/tcp, 0.0.0.0:4194->4194/tcp   cadvisor                                                                                      
```
Interesting thing is, that I have not pulled ```kubernetes/pause``` image on ```node-02```, but after doing it so for ```node-01```, containers started also on this machine.
```
core@node-02 ~ $ docker ps
CONTAINER ID        IMAGE                    COMMAND                CREATED             STATUS              PORTS                              NAMES
89b5e595eddb        kubernetes/pause:go      "/pause"               11 minutes ago      Up 11 minutes                                          k8s_POD.c57df10a_elasticsearch-master-fplln_default_7f1bfb37-ce0a-11e4-b598-080027ddceed_1de31ed6
d8a487946d43        google/cadvisor:0.10.1   "/usr/bin/cadvisor -   15 minutes ago      Up 15 minutes       8080/tcp, 0.0.0.0:4194->4194/tcp   cadvisor                                                       
core@node-02 ~ $ docker ps
CONTAINER ID        IMAGE                        COMMAND                CREATED             STATUS              PORTS                              NAMES
5cb570ac2e55        pires/elasticsearch:master   "/usr/bin/runsvdir -   2 minutes ago       Up 2 minutes                                           k8s_elasticsearch-master.7f022321_elasticsearch-master-fplln_default_7f1bfb37-ce0a-11e4-b598-080027ddceed_d40f2694
89b5e595eddb        kubernetes/pause:go          "/pause"               14 minutes ago      Up 14 minutes                                          k8s_POD.c57df10a_elasticsearch-master-fplln_default_7f1bfb37-ce0a-11e4-b598-080027ddceed_1de31ed6
d8a487946d43        google/cadvisor:0.10.1       "/usr/bin/cadvisor -   19 minutes ago      Up 19 minutes       8080/tcp, 0.0.0.0:4194->4194/tcp   cadvisor                                                                                  
```

**Comment by AntonioMeireles** [comment]
> Interesting thing is, that I have not pulled kubernetes/pause image on node-02,
> but after doing it so for node-01, containers started also on this machine.  

@jdojff well, this, at least, makes perfect sense; given that your docker pulls were going via the registry-cache, once an image is downloaded/pulled from one minion it gets available 'locally' for all others in the cluster. 

>First thing to get my attention is, when starting vagrant up master and vagrant up I get Could not open a >connection to your authentication agent. in each node: http://pastebin.com/Sz8uHt98 Not sure if this has >something to do with it or not.

and things start to make [some] sense. you are running windows right ? the vagrant trigger scripts we run assume that the host is either linux or osx, so it's expected that they will fail. anyway that specific message, regarding the *connection agent* only means that there is not ssh port forwarding between the nodes themselves and should not be related with main issue.

**Comment by jdojff** [comment]
Yes, I'm on Windows. But I can't run vagrant inside of virtualbox linux VM to test if it works..

**Comment by AntonioMeireles** [comment]
@jdoff main issue atm that crosses my mind is if you are having issues *just* because on Windows, or if there's some general issue that persists...

**Comment by pires** [comment]
> First thing to get my attention is, when starting vagrant up master and vagrant up I get Could not open a connection to your authentication agent. in each node: http://pastebin.com/Sz8uHt98 Not sure if this has something to do with it or not.

You need [`ssh-agent` running](http://ryanlanciaux.github.io/blog/2014/05/15/running-ssh-agent-on-windows/).

**Comment by jdojff** [comment]
Yes, ssh-agent solved vagrant startup issue. However, original issue still exists..

**Comment by AntonioMeireles** [comment]
@jdojff it seems that all releases above 0.12.2 are having issues. we're working to spot the root cause of it)

**Comment by pires** [comment]
Windows will not be supported, sorry. Will have this in the documentation.

**Comment by pires** [comment]
So @Hikariii is helping on this!

**Comment by pires** [comment]
Fixed by #81.
